lib/change_stream.js

```diff
@@ -0,0 +1,283 @@
+      });
+    }
+  });
+
+};
+
+// Create a new change stream cursor based on self's configuration
+var createChangeStreamCursor = function (self) {
+  var changeStreamCursor;
+
+  if (self.resumeToken) {
+    self.options.resumeAfter = self.resumeToken;
+  }
+
+  // Create the Change Stream cursor (either using the server or the simulator)
+  var serverSupportsChangeNotifications = true;
+  if (serverSupportsChangeNotifications) {
+    changeStreamCursor = buildChangeStreamAggregationCommand(self.serverConfig, self.namespace, self.pipeline, self.resumeToken, self.options);
+  } else {
+    // Create the change stream cursor using the black box simulator
+    changeStreamCursor = changeStreamSimulator.createChangeStreamCursorSimulation(self.serverConfig, self.namespace, self.pipeline, self.options);
+  }
+
+  /**
+   * Fired for each new matching change in the specified namespace. Attaching a 'data' event listener to a Change Stream will switch the stream into flowing mode. Data will then be passed as soon as it is available.
+   *
+   * @event ChangeStream#data
+   * @type {object}
+   */
+  if (self.listenerCount('data') > 0) {
+    changeStreamCursor.on('data', function (change) {
+      processNewChange(self, null, change);
+    });
+  }
+
+  /**
+   * Change stream close event
+   *
+   * @event ChangeStream#close
+   * @type {null}
+   */
+  changeStreamCursor.on('close', function () {
+    self.emit('close');
+  });
+
+  /**
+   * Change stream end event
+   *
+   * @event ChangeStream#end
+   * @type {null}
+   */
+  changeStreamCursor.on('end', function () {
+    self.emit('end');
+  });
+
+  /**
+   * Fired when the stream encounters an error.
+   *
+   * @event ChangeStream#error
+   * @type {Error}
+   */
+  changeStreamCursor.on('error', function(error) {
+    self.emit('error', error);
+  });
+
+  return changeStreamCursor;
+};
+
+var buildChangeStreamAggregationCommand = function (serverConfig, namespace, pipeline, resumeToken, options) {
+  var changeNotificationStageOptions = {};
+  if (options.fullDocument) {
+    changeNotificationStageOptions.fullDocument = options.fullDocument;
+  }
+  if (resumeToken || options.resumeAfter) {
+    changeNotificationStageOptions.resumeAfter = resumeToken || options.resumeAfter;
+  }
+
+  // Map cursor options
+  var cursorOptions = {};
+  ['maxAwaitTimeMS', 'collation', 'readPreference'].forEach(function(optionName) {
+    if (options[optionName]) {
+      cursorOptions[optionName] = options[optionName];
+    }
+  });
+
+  var changeStreamPipeline = [{$changeNotification: changeNotificationStageOptions}].concat(pipeline);
+  var command = {
+    aggregate : namespace.collection ? namespace.collection : 1,
+    pipeline : changeStreamPipeline,
+    cursor: {
+      batchSize: options.batchSize || 1
+    }
+  };
+
+  // Create and return the cursor
+  return serverConfig.cursor(namespace.database + '.' + namespace.collection, command, cursorOptions);
```

Comment:
sorry if you already answered this, but it looks like you would still potentially be creating a cursor called `1.` if you wanted a change stream on an entire database?